src/testing/debugging.md

```diff
@@ -8,275 +8,26 @@ description: How to debug your app using the DevTools suite.
   support a built-in source-level debugger with
   the ability to set breakpoints, step through code,
   and examine values.
+* If you are looking for a way to use GDB to remotely debug the
+  Flutter engine running within an Android app process,
+  check out [`flutter_gdb`][].
+* [DevTools][], a suite of performance and profiling
+  tools that run in a browser.
 * [Flutter inspector][], a widget inspector available
   in DevTools, and also directly from Android Studio
   and IntelliJ (enabled with the Flutter plugin).
   The inspector allows you to examine a visual
   representation of the widget tree, inspect
   individual widgets and their property values,
   enable the performance overlay, and more.
-* If you are looking for a way to use GDB to remotely debug the
-  Flutter engine running within an Android app process,
-  check out [`flutter_gdb`][].
-
-[`flutter_gdb`]: https://github.com/flutter/engine/blob/main/sky/tools/flutter_gdb
-
-## DevTools
-
-For debugging and profiling apps, DevTools might be
-the first tool you reach for. DevTools runs in a
-browser and supports a variety of features:
-
-* source-level debugger
-* widget inspector that displays a visual widget tree,
-  and "widget select" mode where you select a widget
-  in the app and it drills down to that widget in
-  the tree
-* memory profiler
-* timeline view that supports tracing, and importing
-  and exporting trace information
-* logging view
-
-If you run your application in [debug mode][] or
-[profile mode][], while it's running you can open
-DevTools in the browser to connect to your app.
-DevTools doesn't work well with an app compiled to
-[release mode][], as the debugging and profiling
-information has been stripped away.
-
-If you use DevTools for profiling, make sure to
-run your application in [profile mode][]. Otherwise,
-the main output that appears on your profile are the
-debug asserts verifying the framework's various invariants
-(see [Debug mode assertions](#debug-mode-assertions)).
-
-![GIF showing DevTools features]({{site.url}}/assets/images/docs/tools/devtools/inspector.gif){:width="100%"}
-
-For more information, see the
-[DevTools][] documentation.
-
-## Setting breakpoints
-
-You can set breakpoints directly in your IDE/editor
-(such as [Android Studio/IntelliJ][] and [VS Code][]),
-in the [DevTools debugger][],
-or [programmatically][breakpoints].
-
-## The Dart analyzer
-
-If you're using a [Flutter enabled IDE/editor][],
-the Dart analyzer is already checking your code
-and looking for possible mistakes.
-
-If you run from the command line,
-test your code with `flutter analyze`.
-
-The Dart analyzer makes heavy use of type annotations that
-you put in your code to help track problems down.
-You are encouraged to use them everywhere (avoiding `var`,
-untyped arguments, untyped list literals, and so on)
-as this is the quickest and least painful way of tracking
-down problems.
-
-For more information, see [Using the Dart analyzer][].
```

Comment:
I think this is still relevant

Reply:
Yes, this is relevant. But it should be incorporated into new content / structure. 
I commented on the issue to take some deleted content from this PR.